server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/BasicAuthenticationStrategy.scala

```diff
@@ -85,26 +85,27 @@ object UserCredential {
       }
     }
   }
+}
+
+case class UserCredential(username: Username, password: String)
+
+class BasicAuthenticationStrategy @Inject()(val usersRepository: UsersRepository,
+                                            val mailboxManager: MailboxManager) extends AuthenticationStrategy {
+
+  override def createMailboxSession(httpRequest: HttpServerRequest): Mono[MailboxSession] = {
+    SFlux.fromStream(() => authHeaders(httpRequest).toScala[Stream])
+      .map(parseUserCredentials)
+      .handle(publishNext)
+      .filter(isValid)
+      .map(_.username)
```

Comment:
I tried it before.MO is: `_.username` dont have any context before, so you dont have any clue to know which kind class of `_` represent.

Reply:
You know it's a UserCredentials turned into a Username

Reply:
Yes ok for me btw. How about we just keep it, if anyone confuse with it we can change it.